tests/integration/defs/stress_test/stress_test.py

```diff
@@ -0,0 +1,941 @@
+
+    @property
+    def request_count_list(self) -> List[int]:
+        """Calculate request count based on concurrency"""
+        # Keep fair amount of request count even when concurrency is low
+        result = []
+        for concurrency in self.concurrency_list:
+            if concurrency <= 128:
+                result.append(128)
+            else:
+                result.append(concurrency * 2)
+        return result
+
+
+def filter_server_output(
+        pipe,
+        pattern_to_exclude=r'INFO: .+ - "POST /v1/completions HTTP/1.1" 200 OK'
+):
+    """
+    Filter function that reads from pipe and writes to stdout,
+    excluding lines that match the given pattern.
+    """
+    pattern = re.compile(pattern_to_exclude)
+    try:
+        for line in iter(pipe.readline, ''):
+            # Print lines that don't match the pattern
+            if not pattern.search(line):
+                print(line, end='', flush=True)
+    except (BrokenPipeError, IOError, ValueError) as e:
+        print_warning(f"Pipe error in filter_server_output: {str(e)}")
+
+
+@contextlib.contextmanager
+def launch_process(cmd, start_new_session=True, filter_pattern=None):
+    """
+    Context manager to handle process execution and filter output.
+
+    Args:
+        cmd: Command list to execute
+        start_new_session: Whether to start the process in a new session
+        filter_pattern: Optional regex pattern to exclude from output
+
+    Yields:
+        The process object
+    """
+    process = None
+    stdout_reader = None
+    stderr_reader = None
+
+    try:
+        # Setup pipes for stdout and stderr
+        process = Popen(
+            cmd,
+            start_new_session=start_new_session,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            bufsize=1,  # Line buffered
+            universal_newlines=True)  # Text mode
+
+        print_info(f"Process started with PID: {process.pid}")
+
+        # Start threads to filter and process output
+        if filter_pattern:
+            stdout_reader = threading.Thread(
+                target=filter_server_output,
+                args=(process.stdout, filter_pattern),
+                daemon=True  # Make sure thread doesn't block program exit
+            )
+            stdout_reader.start()
+
+            stderr_reader = threading.Thread(
+                target=filter_server_output,
+                args=(process.stderr, filter_pattern),
+                daemon=True  # Make sure thread doesn't block program exit
+            )
+            stderr_reader.start()
+
+        yield process
+    finally:
+        if process:
+            print_info(f"Stopping process with PID: {process.pid}")
+            if process.poll() is None:
+                # Send termination signal
+                process.terminate()
+
+                try:
+                    process.wait(timeout=GRACEFUL_TERMINATION_TIMEOUT)
```

Comment:
This terminate will just exit the process with finishing to on-the-fly tokens or not?
If all the token generation has to be finished, 4 seconds may not be enough for long osl cases.

Reply:
It first wait for process kill signal, e.g., trtllm-serve or genai-perf, then wait 4 seconds for the process to exit cleanly, if not then will be forced to kill the process and its subprocess cleanly (use `cleanup_process_tree`), I know your point, 4 seconds is good enough for tinyllama case, for larger model like v3 we can adjust this value based some pre-run experiments.

Reply:
> with finishing to on-the-fly tokens
This piece of code is to stop `trtllm-serve` after all test finished or something went wrong, which means there's either no on-the-fly requests or we merely want to clean up all processes in whatever manner.

> may not be enough for long osl cases
`genai-perf` will wait all tokens to be generated before quit, so when reaching this line, `trtllm-serve` has completed generating all tokens for all requests, and is idling.